### PR TITLE
Weak segments + label and prediction drift

### PR DESCRIPTION
**Pull request by Nadav-Barak**
Added checks - currently without tests
Mechanism for suppling metadata
Additional small stuff

### CODE REVIEW

File: deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        self.n_samples = n_samples
+        self.n_to_show = n_to_show
+        self.loss_per_sample = loss_per_sample
+        self.alternative_scorer = alternative_scorer if alternative_scorer else None
+        self.categorical_aggregation_threshold = categorical_aggregation_threshold
+
+    def run_logic(self, context: Context, dataset_kind) -> CheckResult:
+        """Run check."""
+        text_data = context.get_data_by_kind(dataset_kind)
+        if text_data.meta_data is None:
```
Comment (nirhutnik) [inline deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py:87]: additional_data pls :) 
and in the errors too

File: deepchecks/nlp/checks/train_test_validation/train_test_prediction_drift.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        CheckResult
+            value: drift score.
+            display: label distribution graph, comparing the train and test distributions.
+        """
+        train_dataset = context.train.sample(self.n_samples, random_state=context.random_state)
+        test_dataset = context.test.sample(self.n_samples, random_state=context.random_state)
+        model = context.model
+
+        drift_score_dict, drift_display_dict = {}, {}
+        method, classes = None, context.model_classes
```
Comment (nirhutnik) [inline deepchecks/nlp/checks/train_test_validation/train_test_prediction_drift.py:139]: Not sure what the "method" here is

File: deepchecks/nlp/text_data.py [hunk 3]
```diff
@@ -84,6 +85,7 @@ def __init__(
             task_type: t.Optional[str] = None,
             dataset_name: t.Optional[str] = None,
             index: t.Optional[t.Sequence[t.Any]] = None,
+            meta_data: t.Optional[pd.DataFrame] = None
```
Comment (nirhutnik) [inline deepchecks/nlp/text_data.py:88]: addiotional_data etc.

File: deepchecks/nlp/text_data.py [hunk 4]
```diff
@@ -107,38 +109,51 @@ def __init__(
...
             if len(raw_text) != len(tokenized_text):
                 raise DeepchecksValueError('raw_text and tokenized_text must have the same length')
 
         if index is None:
-            index = np.arange(len(raw_text))
+            self.index = list(range(len(raw_text)))
         elif len(index) != len(raw_text):
             raise DeepchecksValueError('index must be the same length as raw_text')
+        else:
+            self.index = list(index)
```
Comment (nirhutnik) [inline deepchecks/nlp/text_data.py:133]: Nirpicking, but why add unnecessary lines of code? was shorter before with less "else" statements and value assigning.

**Review by nirhutnik — APPROVED** [review]

File: deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py (hunk 1 above)
Comment (Nadav-Barak) [reply deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py:87]: Right!

File: deepchecks/nlp/checks/train_test_validation/train_test_prediction_drift.py (hunk 2 above)
Comment (Nadav-Barak) [reply deepchecks/nlp/checks/train_test_validation/train_test_prediction_drift.py:139]: Me neither, removed

File: deepchecks/nlp/text_data.py [hunk 5]
```diff
@@ -244,30 +256,48 @@ def sample(self: TDataset, n_samples: int, replace: bool = False, random_state:
...
+        Parameters
+        ----------
+        index : int
+            Index of sample to get.
+        Returns
+        -------
+        str
+            Raw text of sample.
+        """
+        return self._text[self.index.index(index)]
```
[the remark this answers is not in the document]
Comment (Nadav-Barak) [reply deepchecks/nlp/text_data.py:273]: my main problem here was too many index in the same line🤣
